src/tests/test_xmss.cpp

```diff
@@ -207,12 +207,122 @@ std::vector<Test::Result> xmss_statefulness()
+      "101A0D3E7DA3152F7D698ED3F6F6EE79B4F87E6032B98907E2FEC3AD1F44FA7816B568A"
+      "31AF329B19636788498BE3C21A5676BDEB987F1BA213C8E86F5D39CEC9571EB5198645C"
+      "D39EA26BE5598D47F3324F3D763D4E892A13C579F964B72369B40800B30088DE6140E94"
+      "5B5F03C67015EC4E03065EC9126C2417F827EC526C852020D73BB6F97DED24081399D7B"
+      "F7EE644B16A7BC601F32A312CFB98FF0E954DF0FCD2C61154B3D8CCC32DCFCCB172033D"
+      "5F5ACB03369E323429123FB9EAB658CB072EA9CD058CAD76D78A6F0BA01BB5D36172DE7"
+      "4DE17704A055509B3BDBE7032F4A0E3E7329CC646D06DE7C7ECF5EF030362CE079D3C3B"
+      "DDB605A04918C16FE79251B7705DAC88A0C84E7C692E1AFF59968DE28CBDF001BBEA509"
+      "A8A270B6BFD7E653B2228418E73F40A14F491FAEBAC8C11F346FB51C8E10DB63ED5C287"
+      "DA7DDC692B8FE6998AC85EA29D4F9B5FEB3E062FFEEA23DAD536DDE783AFE43606D9685"
+      "09F4695E55FE935DF20B8E9EFEE01DD21E733C22830A6E6086EBB4AC7CFB5DC5C6F9FBE"
+      "F131BF636A129723E5D663FDCCF181AB996D158146602D567B48CEF809E4D3E5B7A7B64"
+      "3C48EFCA7621534397B623AF23AE6F2BE7F2D7A6FBB2D7B64B96526B73DBA66A29F4260"
+      "2BD9FF72A165431C15A570A5731D9426210F946FC682F181944F40DFB68314ACA43E321"
+      "4AFDF11EA825B79425F0C864E354CA7E6C447AE9E0796AAB2B6720516D3FC0B73CFE98E"
+      "F0D8E8D80A59DB9ADA0F03CF0444E1BED9ED9A75FBB6FD6D5BAD4F5FF66C8987ADA6F9F"
+      "CA06CBAD1C9E329B844CC5BBF6CCC52852DE44B72EECAF34802043107542F5BFEA73944"
+      "611AB035D6D7A8EBD3101F321A54C1DADC600419F8DA6BF3967E5B9E708BC4C3E901623"
+      "E8FB6E87825757498A24A921EC601CC15399237D9C4115B678E8968A2095818D9261C4E"
+      "8F94B9DCE3F74EC7E0479F6F6AB13676954390EF3B45D2292BF664D618CFE5ECFFBD391"
+      "872B4F9F306A09B1238E7561942C7065FD14208F4D4A42752A9A747A29B8BE3C1658EDC"
+      "6B250DBD1599FBB09A7F148A7AEFEAB26ADB728A330DD3F616C8A736D1BF4EA17F2C3BF"
+      "A5E22C249FA9D1E7DA08DB351709C4");
+
+   Botan::XMSS_PrivateKey secret_key = Botan::XMSS_PrivateKey(legacy_xmss_private_key, Botan::WOTS_Derivation_Method::Botan2x);
+   Botan::XMSS_PublicKey public_key(secret_key);
+
+   const auto message = Botan::hex_decode("deadcafe");
+   const auto algo_name = "SHA2_10_256";
+
+   return
+      {
+      Botan_Tests::CHECK("Use a legacy private key to create a signature", [&](auto& result)
+         {
+         Botan::PK_Signer signer(secret_key, Test::rng(), algo_name);
+         auto signature = signer.sign_message(message, Test::rng());
+
+         Botan::PK_Verifier verifier(public_key, algo_name);
+         result.confirm("legacy private key generates signatures that are still verifable",
+                        verifier.verify_message(message, signature));
+         }),
+
+      Botan_Tests::CHECK("Verify a legacy signature", [&](auto& result)
+         {
+         Botan::PK_Verifier verifier(public_key, algo_name);
+         result.confirm("legacy private key generates signatures that are still verifable",
+                        verifier.verify_message(message, legacy_signature));
+         }),
```

Comment:
Maybe we should add another test that verifies a legacy signature with a `Public_Key` object that was obtained by straight from a legacy public key. This particular `public_key` was obtained as the base class of the legacy private key.

I.e. have  another public key that is instantiated like so:

```C++
const auto legacy_public_key = Botan::XMSS_PublicKey(Botan::hex_decode("04818800000001C49B128365E23D97C2D36A2AC26E0B448FCEF77A5ED68EFDC460E787D0EDB6768EAC3E82B04B84571C49684ED822F8179FDFF1C825DA9F9EA78F5D50A21E6341");
```

and create a third `CHECK`-context that verifies a new signature (created by the legacy private key) with the `legacy_public_key`. My main objective: Make sure that the `WOTS_Derivation_Method::Botan2x` used in the creation of the existing `public_key` object did not have a side effect. I.e. even if no such extra step is involved, the validation still works.

We can do that in pair-programming, if you want. Not sure whether I made myself clear enough, tbh. 😄

Reply:
I think I understood your suggestion. Let's discuss tomorrow, how to procede :) 
Thanks for the review btw!